internal/pkg/agent/application/monitoring/server_test.go

```diff
@@ -5,11 +5,89 @@
 package monitoring
 
 import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
 	"testing"
+	"time"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent-libs/api"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/coordinator"
+	"github.com/elastic/elastic-agent/internal/pkg/core/monitoring/config"
+	"github.com/elastic/elastic-agent/pkg/component"
+	"github.com/elastic/elastic-agent/pkg/component/runtime"
 )
 
+func TestConfigUpdateOnReload(t *testing.T) {
```

Comment:
Am I blind or is this test just starting the server and not actually reloading anything?

Reply:
Yeah, the name is a tad deceptive, the integration test is the only one doing the reloading.